src/components/SwapWidget/SwapWidget.tsx

```diff
@@ -0,0 +1,247 @@
+import { selectBalances } from "../../features/balances/balancesSlice";
+import { Web3Provider } from "@ethersproject/providers";
+import { useWeb3React } from "@web3-react/core";
+import { useMatomo } from "@datapunt/matomo-tracker-react";
+import { useTranslation } from "react-i18next";
+import Button from '../Button/Button';
+import TokenSelect from '../TokenSelect/TokenSelect';
+import Timer from "../../components/Timer/Timer";
+import Modal from "react-modal";
+import Card from '../Card/Card';
+import WalletProviderList from '../WalletProviderList/WalletProviderList';
+
+const floatRegExp = new RegExp("^[+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)$");
+
+const SwapWidget = () => {
+  const [senderToken, setSenderToken] = useState<string>();
+  const [signerToken, setSignerToken] = useState<string>();
+  const [senderAmount, setSenderAmount] = useState("0.01");
+  const [isConnecting, setIsConnecting] = useState<boolean>(false);
+  const [showWalletList, setShowWalletList] = useState<boolean>(false);
+  const transactions = useAppSelector(selectTransactions);
+  const balances = useAppSelector(selectBalances);
+  const order = useAppSelector(selectBestOrder);
+  const ordersStatus = useAppSelector(selectOrdersStatus);
+  const dispatch = useAppDispatch();
+  const activeTokens = useAppSelector(selectActiveTokens);  
+  const { t } = useTranslation(["orders", "common", "wallet"]);
+  const { chainId, account, library, active } = useWeb3React<Web3Provider>();
+  const { trackEvent } = useMatomo();
+
+    
+
+  const getTokenDecimals = (tokenAddress: string) => {
+    for (const token of activeTokens) {
+      if (token.address === tokenAddress) return token.decimals;
+    }
+    return 18;
+  };
+
+  const getTokenApprovalStatus = (tokenId: string | undefined) => {
+    if (tokenId === undefined) return null;
+    for (let i = 0; i < transactions.length; i++) {
+      if (transactions[i].type === "Approval") {
+        const approvalTx: SubmittedApproval = transactions[
+          i
+        ] as SubmittedApproval;
+        if (approvalTx.tokenAddress === tokenId) return approvalTx.status;
+      }
+    }
+    return null;
+  };
+
+  // function to only allow numerical and dot values to be inputted
+  const handleTokenAmountChange = (e: FormEvent<HTMLInputElement>) => {
+    const value = e.currentTarget.value;
+    if (value === "" || floatRegExp.test(value)) {
+      setSenderAmount(value);
+    }
+  };
+
+  const DisplayedButton = () => {
+    if (!active || !chainId) {
+      return (
+        <Button
+          className="w-full mt-2"
+          intent="primary"
+          disabled={!senderToken || !signerToken || !senderAmount}
+          loading={isConnecting}
+          onClick={() => setShowWalletList(true)}
+        >
+        {t("wallet:connectWallet")}
+      </Button>
+      )
+    } else if (
+      signerAmount && 
+      getTokenApprovalStatus(senderToken) === "succeeded" &&
```

Comment:
Caught this one myself, shouldn't use `getTokenApprovalStatus` here, should probably write some function to check the `senderAmount` against an explicit ERC20 approval balance check. Became apparent this line wasn't a great idea because I had merged in develop branch changes and I think we had made a change to the localStorage key so things were breaking 🥴

Reply:
Good point, I'll get on it. And yes we're doing unlimited allowances, which I prefer. I remember finding it convenient not having to constantly approve when I was trading on Uniswap heavily last summer.